build-tools/src/main/java/org/coner/checks/NeedBraces62Check.java

```diff
@@ -0,0 +1,145 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2014  Oliver Burn
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.coner.checks;
+
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks for braces around code blocks.
+ * </p>
+ * <p> By default the check will check the following blocks:
+ *  {@link TokenTypes#LITERAL_DO LITERAL_DO},
+ *  {@link TokenTypes#LITERAL_ELSE LITERAL_ELSE},
+ *  {@link TokenTypes#LITERAL_FOR LITERAL_FOR},
+ *  {@link TokenTypes#LITERAL_IF LITERAL_IF},
+ *  {@link TokenTypes#LITERAL_WHILE LITERAL_WHILE}.
+ * </p>
+ * <p>
+ * An example of how to configure the check is:
+ * </p>
+ * <pre>
+ * &lt;module name="NeedBraces"/&gt;
+ * </pre>
+ * <p> An example of how to configure the check for <code>if</code> and
+ * <code>else</code> blocks is:
+ * </p>
+ * <pre>
+ * &lt;module name="NeedBraces"&gt;
+ *     &lt;property name="tokens" value="LITERAL_IF, LITERAL_ELSE"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * Check has an option <b>allowSingleLineIf</b> which allows one line
+ * if-statements without braces, e.g.:
+ * <p>
+ * <code>
+ * if (obj.isValid()) return true;
+ * </code>
+ * </p>
+ * <br>
+ *
+ * @author Rick Giles
```

Comment:
Where did you find the original source for this file? If we can't include it as a maven dependency, I'd at least like it to have a link to the original.